### DISCUSSION TITLE
Aggregate elastic tensor for multi-phase systems

### DISCUSSION THREAD

**Discussion by matthiaskrug7** [mtex-toolbox/mtex#1099] [Ask Anything]
Hi,

I have a polycrystalline aggregate consisting of 3 phases A (7 vol%), B (33 vol%) and C (60 vol%). For each of the phases, I know the single-crystal elastic tensor and I know its texture, i.e. I have a list of Euler angles for each phase. Now, I would like to know the seismic properties of the aggregate. 

I know how to calculate the seismic properties for an individual phase based on the Euler angles and the single-crystal tensor, but simply calculating a weighted average of a property to obtain the aggregate property doesn´t seem correct to me. I think the correct way would be to generate an elastic tensor of the whole aggregate and to compute the seismic properties from this aggregate elastic tensor. 

Does anyone know how to combine several elastic tensors with taking the respective volume proportions of each phase into account?

**Comment by lfgmorales** [comment]
Hi


DId you look here?


https://mtex-toolbox.github.io/CPOSeismicProperties.html


https://mtex-toolbox.github.io/TensorAverage.html



cheers



Luiz


Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________
From: matthiaskrug7 ***@***.***>
Sent: 15 June 2021 14:24:15
To: mtex-toolbox/mtex
Cc: Subscribed
Subject: [mtex-toolbox/mtex] Aggregate elastic tensor for multi-phase systems (#1099)


Hi,

I have a polycrystalline aggregate consisting of 3 phases A (7 vol%), B (33 vol%) and C (60 vol%). For each of the phases, I know the single-crystal elastic tensor and I know its texture, i.e. I have a list of Euler angles for each phase. Now, I would like to know the seismic properties of the aggregate.

I know how to calculate the seismic properties for an individual phase based on the Euler angles and the single-crystal tensor, but simply calculating a weighted average of a property to obtain the aggregate property doesn´t seem correct to me. I think the correct way would be to generate an elastic tensor of the whole aggregate and to compute the seismic properties from this aggregate elastic tensor.

Does anyone know how to combine several elastic tensors with taking the respective volume proportions of each phase into account?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1099>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN73QBRR5N2BCWOACUTTS5A67ANCNFSM46XFDB3A>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1099", "url": "https://github.com/mtex-toolbox/mtex/discussions/1099", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by matthiaskrug7** [reply]
Hi Luiz,

Thanks for your quick reply! I checked out those pages before but missed the part that an EBSD object is in fact not necessary. I tried it out and it seems to work. Thanks a lot!

Best,
Matthias

**Comment by smerkel** [comment]
Hi Luiz,

I have an issue with the formula `CHill = vol_ol * CHill_ol + vol_opx * CHill_opx + vol_cpx * CHill_cpx` in the following page https://mtex-toolbox.github.io/CPOSeismicProperties.html

I do not think that it is completely correct.

1. First, the density that is assigned to CHill should be the weighted average of the density of ol, cpx, and opx. In test we ran we Matthias here, it is not what is being calculated. MTeX take the density assigned to the first tensor (ol in the above example).
2. Second, when we test the formula, MTeX  simply calculates a weighted average of the Cij's. I do not think this is physically correct in all cases.

If the wavelength is larger that the grain size, all grains will be under similar strain as the wave comes in, this is a Voigt average, in which you make the average of the Cij, which is what is done in MTeX. In this case, we just need to correct the calculation of the average density in MTeX is it should be ok.

If the wavelength is smaller that the grain size, the wave will travel from one grain to the next, the travel time will be the sum of the travel times. The velocity will be the weighted average of the velocities of each phase, which is not what it is being calculated with this formula.

In geosciences, the wavelength is probably always larger that the grain size, but we might want to write a note about this in the MTeX manual.

Sébastien

**Comment by lfgmorales** [comment]
Hey Sébastien



Good to hear from you, and the comments are really appreciated, I will check the poitns you mentioned and come back to you soon


One point that you are right and I will add a note on the is the fact that when a wave propagate through an aggregate (and assuming that the wavelenght is larger than the grain size), the grains will have a similar elastic strain. However, the same wave not only propagate through "grains" from where we know the elastic constants, but through everything that is in there, including grain boundaries, potential porosity, potential layering and so on, but when we do the calculations only assuming the a CPO + the elastic constants, we are only considering one of these terms. Of course in lower mantle materials like the ones you study you don't expect much effect of porosity, etc, but you still have grain boundaries, that in certain cases have a pretty decent effect on the bulk seismic properties. As you know better than I, the Hill average has not really a physical explanation, and it is used up to now because it gives a decent approximation with experimental measurements, but it doesnt mean it is physically correct. I am attaching two graphics that you may have seen before, related to Vp average Hill and anisotropy of some single crystals



cheers



Luiz








Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Comment by smerkel** [comment]
Hi Luiz,

In fact grain boundaries and porosity will have an effect as well. It is probably very hard to model properly.

My comment on density still holds. The operation  `Cag = vol_ol * C_ol + vol_opx * C_opx` does not return the proper density in `Cag`.

Regarding Hill / Reuss / Voigt averages, 

- at present when you write `Cag = vol_ol * C_ol + vol_opx * C_opx`, if you look at the numbers, MTeX calculates a weighted average of the elastic constants Cij
- a weighted average of the Cij is effectively assuming constrain strain -> it is a Voigt average (this how you calculate the Voigt mean using an ODF, you take the mean of the Cij weighted by the orientation probability)
- a weighted average of the Sij is effectively assuming constrain stress -> it is a Reuss average (this how you calculate the Reuss mean using an ODF, you take the mean of the Sij weighted by the orientation probability)

If you want a true Hill average, you would need to
- calculate the Reuss mean with the Cij `CagVoigt = vol_ol * C_ol + vol_opx * C_opx`
- calculate the Voigt mean with the Sij `SagReuss = vol_ol * S_ol + vol_opx * S_opx` (I do not know how to calculate a compliance tensor in MTex, but there might be a way)
- take the average `CHil = (CagVoigt + 1 / SagReuss)/2`

Other other words, a simple mean of elastic constants for two phases is a Voigt average of the aggregate properties, even if the individual elastic tensors of each phase is calculated with a Hill average!

Best,

Sébastien

**Comment by lfgmorales** [comment]
Hi Sébastien



Thanks for your careful into that, I will  have to look into the functions used for those calculations to make they do what they are suppose to do, and then adjust the help, your comments are highly appreciated and I apologize if the help files are not 100% correct


cheers



Luiz





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Comment by ralfHielscher** [comment]
Hi Sebastien,

you are absolutely right in both issues and big kudos to you for pointing this out. We did the following:

1. ``` Cag = vol_ol * C_ol + vol_opx * C_opx``` will raise a warning that the density is not transferred during this calculation and the user needs to set it manually using ```Cag.density = (vol_ol * C_ol.density + vol_opx * C_opx.density) / (vol_ol + vol_opx)```

2. We advice now in out documentation to compute the average bulk tensor by 
 
```matlab
[CVag,CRag,CHag] = mean([C_ol, C_opx ], 'weights', [vol_ol vol_opx], 'density', bulkDensity )
```
which will do the correct thing. Maybe we will even include the right density computation into this mean function.

Thank you very much,

Ralf.

**Comment by smerkel** [comment]
Hi Ralf,

Sorry for the delay. If it was up to me, I would include the proper density into the mean function. I am afraid that many people would get this wrong, students in particular.

This is critical when people calculate seismic velocities. They will be wrong if you do not have the proper density.

Best,

Sébastien

PS: I was also thinking of what would be the best way to treat multiphase materials regarding wave propagation but I am not sure myself. Simple Reuss / Voigt / Hill average have to be treated with cautions, but I would have to converge myself before suggesting any change of a procedure!

**Reply by ralfHielscher** [reply]
Hi Sebastien,

I completely agree with your. Because of this, we propose now in the documentation a more safe procedure
```matlab
vol_ol  = length(ebsd('f')) ./ length(ebsd('indexed'));
vol_opx = length(ebsd('e')) ./ length(ebsd('indexed'));
vol_cpx = length(ebsd('d')) ./ length(ebsd('indexed'));

[CVoigt, CReuss, CHill] = mean([CVoigt_ol, CVoigt_opx, CVoigt_cpx], 'weights',[vol_ol, vol_opx, vol_cpx]);
```
This utilizes the tensor/mean function that computes not only the correct averages with respect to given volume percentages, but also 
assigns the correct density to them.

What do you think about this?

Ralf.

**Reply by smerkel** [reply]
This looks good to me!

Best,

Sébastien
